docs/guides/Weni-by-VTEX/CLI/core-concepts.md

```diff
@@ -0,0 +1,657 @@
+
+### Credentials in Production Environment
+
+When your agent is deployed on the Weni by VTEX Platform, credentials are securely managed by the system. This ensures that sensitive information, such as API keys and access tokens, is stored and transmitted securely.
+
+#### How to Configure Credentials for Production
+
+1. **Define credentials in the YAML file**: Specify all necessary credentials in the `credentials` section of your agent definition file.
+
+2. **Deploy your agent**: When pushing your agent to the Weni by VTEX Platform using the CLI, the system will automatically detect the credentials defined in your YAML file.
+
+3. **Configure values in the interface**: Administrators will be able to configure the actual credential values through the Weni by VTEX Platform interface, without needing to modify the code.
+
+4. **Associate credentials with tools**: Ensure that each tool that needs credentials is correctly configured to access them through the `context.credentials` object.
+
+> **Note**: When you assign your agent in the Weni by VTEX Platform, the credentials defined in your YAML file will be displayed in the interface for configuration. For example, if you have the following agent definition:
+> 
+> ```yaml
+> agents:
+>     cep_agent:
+>         credentials:
+>             api_key:
+>                 label: "API Key"
+>                 placeholder: "Enter your API key"
+>         name: "CEP Agent"
+>         description: "Weni's CEP agent with components"
+>         instructions:
+>         - "You are an expert in providing addresses to the user based on a postal code provided by the user"
+>         - "The user will send a ZIP code (postal code) and you must provide the address corresponding to this code."
+>         guardrails:
+>         - "Don't talk about politics, religion or any other sensitive topic. Keep it neutral."
+>         tools:
+>         - get_address:
+>             name: "Get Address"
+>             source:
+>                 path: "tools/get_address"
+>                 entrypoint: "main.GetAddress"
+>                 path_test: "test_definition.yaml"
+>             description: "Function to get the address from the postal code"
+>             parameters:
+>                 - cep:
+>                     description: "postal code of a place"
+>                     type: "string"
+>                     required: true
+>                     contact_field: true
+> ```
+> 
+> After running the command `weni project push agent_definition.yaml` to upload your agent to your project on the Weni by VTEX Platform, you'll find your agent in the Agent Builder gallery. When you select and assign this agent, you'll be prompted to enter the actual values for the credentials you defined (in this case, the API Key).
+
+
+> ![Agents Gallery](assets/agent-builder-gallery.png)
+
+> Enter you credentials:
+
+> ![Assign Agent](assets/agent-builder-assign.png)
```

Comment:
This path won't work in our portal. You have to add the Github hosted image URL.